vid/src/avidm_gf2.rs

```diff
@@ -0,0 +1,337 @@
+//! This module implements the AVID-M scheme over GF2
+
+use std::ops::Range;
+
+use anyhow::anyhow;
+use jf_merkle_tree::{hasher::HasherNode, MerkleTreeScheme};
+use jf_utils::canonical;
+use serde::{Deserialize, Serialize};
+use sha2::Digest;
+
+use crate::{VidError, VidResult, VidScheme};
+
+/// Namespaced AvidMGF2 scheme
+pub mod namespaced;
+/// Namespace proofs for AvidMGF2 scheme
+pub mod proofs;
+
+/// Merkle tree scheme used in the VID
+pub type MerkleTree =
+    jf_merkle_tree::hasher::HasherMerkleTree<sha3::Keccak256, HasherNode<sha3::Keccak256>>;
+type MerkleProof = <MerkleTree as MerkleTreeScheme>::MembershipProof;
+type MerkleCommit = <MerkleTree as MerkleTreeScheme>::Commitment;
+
+/// Dummy struct for AVID-M scheme over GF2
+pub struct AvidMGF2Scheme;
+
+/// VID Parameters
+#[derive(Clone, Debug, Hash, Serialize, Deserialize, PartialEq, Eq)]
+pub struct AvidMGF2Param {
+    /// Total weights of all storage nodes
+    pub total_weights: usize,
+    /// Minimum collective weights required to recover the original payload.
+    pub recovery_threshold: usize,
+}
+
+impl AvidMGF2Param {
+    /// Construct a new [`AvidMGF2Param`].
+    pub fn new(recovery_threshold: usize, total_weights: usize) -> VidResult<Self> {
+        if recovery_threshold == 0 || total_weights < recovery_threshold {
+            return Err(VidError::InvalidParam);
+        }
+        Ok(Self {
+            total_weights,
+            recovery_threshold,
+        })
+    }
+}
+
+/// Share type to be distributed among the parties.
+#[derive(Clone, Debug, Hash, Serialize, Deserialize, Eq, PartialEq)]
+pub struct RawAvidMGF2Share {
+    /// Range of this share in the encoded payload.
+    range: Range<usize>,
```

Comment:
1. "Share type to be distributed among the parties." is misleading? i think people should distribute `AvidmGf2Share`? 
2. should `range` be a metadata on the `Share` instead of the `RawShare` here?